### PR TITLE
build: experiment with initial implementation of summary check

### PR DESCRIPTION
**Pull request by sofisl**
This is an experimental workflow related to this PRD: https://docs.google.com/document/d/1iUXmYZ5bjXOgXlBQ8mv_qyvegEPkVHyww0RdBKzOMu8/edit?usp=sharing

This is an initial/lightweight prototype of the PRD. It is not meant to be scalable.

### CODE REVIEW

File: .github/workflows/summary_check.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  summary_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: summary_check
+        uses: sofisl/checkRunAction@v10
+        with:
+          conclusion: ${{ github.event.status.state }}
+          pull_request_head_sha: ${{ github.event.status.sha }}
+          author: ${{ github.actor }}
+          myToken: ${{ github.token }}
```
Comment (bcoe) [inline .github/workflows/summary_check.yaml:16]: I think this should be `secrets.GITHUB_TOKEN` then it's magically populated for you.

File: .github/workflows/summary_check.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# This workflow is triggered on check runs
+on: 
+  status
+
+jobs:
+  summary_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: summary_check
+        uses: sofisl/checkRunAction@v10
```
Comment (bcoe) [inline .github/workflows/summary_check.yaml:11]: nit, but when  we  move this to  its final place I would probably use hyphens, `check-run-action`.

File: .github/workflows/summary_check.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        with:
+          conclusion: ${{ github.event.status.state }}
+          pull_request_head_sha: ${{ github.event.status.sha }}
+          author: ${{ github.actor }}
+          myToken: ${{ github.token }}
+          owner: ${{ github.repository_owner }}
+          repo: ${{ github.repository }}
+          testName: ${{ github.event.status.context }}
+          testNameToCheckAgainst: 'ci/kokoro: Samples test'
```
Comment (bcoe) [inline .github/workflows/summary_check.yaml:20]: I see what you did here, do you  think we'd run a copy of this action for system tests and for sample tests?

Comment (sofisl) [reply .github/workflows/summary_check.yaml]: Yes, exactly!

**Pull request closed without merging**